mycore-neo4j/src/main/java/org/mycore/mcr/neo4j/datamodel/metadata/neo4jparser/MCRNeo4JParser.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * This file is part of ***  M y C o R e  ***
+ * See http://www.mycore.de/ for details.
+ *
+ * MyCoRe is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MyCoRe is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MyCoRe.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.mycore.mcr.neo4j.datamodel.metadata.neo4jparser;
+
+import static org.mycore.mcr.neo4j.datamodel.metadata.neo4jutil.MCRNeo4JConstants.NEO4J_CONFIG_PREFIX;
+import static org.mycore.mcr.neo4j.utils.MCRNeo4JUtilsConfigurationHelper.getConfiguration;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jdom2.Attribute;
+import org.jdom2.Document;
+import org.jdom2.Element;
+import org.jdom2.filter.Filters;
+import org.jdom2.xpath.XPathExpression;
+import org.jdom2.xpath.XPathFactory;
+import org.mycore.common.MCRException;
+import org.mycore.common.config.MCRConfiguration2;
+import org.mycore.datamodel.metadata.MCRObject;
+import org.mycore.datamodel.metadata.MCRObjectID;
+import org.mycore.mcr.neo4j.datamodel.metadata.neo4jutil.Neo4JNode;
+import org.mycore.mcr.neo4j.datamodel.metadata.neo4jutil.Neo4JRelation;
+
+/**
+ * Parser Class for Neo4J
+ * @author Andreas Kluge (ai112vezo)
+ */
+public class MCRNeo4JParser implements MCRNeo4JMetaParser {
+
+   private static final Logger LOGGER = LogManager.getLogger(MCRNeo4JParser.class);
+   private static final String LANG_UNSET = "__unset__";
+   private final Map<String, MCRNeo4JAbstractDataModelParser> parserMap;
+
+   public MCRNeo4JParser() {
+      Map<String, String> propertiesMap = MCRConfiguration2.getSubPropertiesMap(NEO4J_CONFIG_PREFIX + "ParserClass.");
```

Comment:
Is the code redundant to map creation code in MCRNeo4JMetaXMLParser?
If yes ...
- move map to a central location,  UtilityClass?
- move and reuse map creation code?

Reply:
Was moved to util class